client/sse.go

```diff
@@ -55,8 +55,16 @@ func NewSSEMCPClient(baseURL string, options ...ClientOption) (*SSEMCPClient, er
 	}
 
 	smc := &SSEMCPClient{
-		baseURL:      parsedURL,
-		httpClient:   &http.Client{},
+		baseURL: parsedURL,
+		//httpClient:   &http.Client{},
+		httpClient: &http.Client{
+			Transport: &http.Transport{
+				MaxIdleConns:        100,
+				MaxIdleConnsPerHost: 10,
+				IdleConnTimeout:     90 * time.Second,
+				DisableKeepAlives:   false,
+			},
+		},
```

Comment:
Might make sense to implement this as a `WithTransport(*http.Transport)` option, rather than hard-coding the settings?